newIDE/app/src/InAppTutorial/InAppTutorialOrchestrator.js

```diff
@@ -961,6 +971,17 @@ const InAppTutorialOrchestrator = React.forwardRef<
                 dialogContent={endDialog}
                 endTutorial={() => {
                   setDisplayEndDialog(false);
+                  if (isRunningMiniTutorial) {
+                    // If running a mini tutorial, we save the progress to indicate that the user has finished this lesson.
+                    preferences.saveTutorialProgress({
+                      tutorialId: tutorial.id,
+                      userId: authenticatedUser.profile
+                        ? authenticatedUser.profile.id
+                        : undefined,
+                      ...getProgress(),
+                      // We do not specify a storage provider, as we don't need to reload the project.
```

Comment:
Safety check: This won't interfere with fling game?

Reply:
Nope, checked!